frontend/src/metabase/components/ErrorPages/ErrorDiagnosticModal.tsx

```diff
@@ -86,24 +83,27 @@ export const ErrorDiagnosticModal = ({
     onClose();
   };
 
-  const handleSlackSubmit = async (values: PayloadSelection) => {
+  const handleSlackSubmit = async (
+    values: Record<string, boolean | string>,
+  ) => {
     setIsSlackSending(true);
-    const selectedKeys = Object.keys(values).filter(
```

Comment:
I would actually put the analytics event tracking in the submit handlers. that will simplify the code changes a good bit.